src/Scroll.php

```diff
@@ -10,6 +10,8 @@
  * @author Manuel Andreo Garcia <andreo.garcia@gmail.com>
  *
  * @see https://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-scroll.html
+ *
+ * @template-implements \Iterator<string|null, ResultSet>
```

Comment:
```suggestion
 * @template-implements \Iterator<string, ResultSet>
```

The iterator key is never `null` when the iterator is in a state where the `valid` method returns `true` (which is the state you care about during iteration)